apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClient.DatasetCrud.cs

```diff
@@ -181,6 +181,86 @@ public virtual void DeleteDataset(DatasetReference datasetReference, DeleteDatas
+            UpdateDataset(GetDatasetReference(projectId, datasetId), resource, options);
+
+        /// <summary>
+        /// Updates the specified dataset within this client's project to match the given resource.
+        /// </summary>
+        /// <remarks>
+        /// If the resource contains an ETag, it is used for optimistic concurrency validation.
+        /// </remarks>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="resource">The dataset resource representation to use for the update. All updatable fields will be updated.</param>
+        /// <param name="options">The options for the operation. May be null, in which case defaults will be supplied.</param>
+        /// <returns>The updated dataset.</returns>
+        public virtual BigQueryDataset UpdateDataset(string datasetId, Dataset resource, UpdateDatasetOptions options = null) =>
+            UpdateDataset(GetDatasetReference(datasetId), resource, options);
+
+        /// <summary>
+        /// Updates the specified dataset to match the given resource.
+        /// </summary>
+        /// <remarks>
+        /// If the resource contains an ETag, it is used for optimistic concurrency validation.
+        /// </remarks>
+        /// <param name="datasetReference">A fully-qualified identifier for the dataset. Must not be null.</param>
+        /// <param name="resource">The dataset resource representation to use for the update. All updatable fields will be updated.</param>
+        /// <param name="options">The options for the operation. May be null, in which case defaults will be supplied.</param>
+        /// <returns>The updated dataset.</returns>
+        public virtual BigQueryDataset UpdateDataset(DatasetReference datasetReference, Dataset resource, UpdateDatasetOptions options = null) =>
+            throw new NotImplementedException();
+
+        /// <summary>
+        /// Patches the specified dataset with fields in the given resource.
+        /// </summary>
+        /// <remarks>
+        /// If the resource contains an ETag, it is used for optimistic concurrency validation.
+        /// </remarks>
+        /// <param name="projectId">The project ID. Must not be null.</param>
+        /// <param name="datasetId">The dataset ID. Must not be null.</param>
+        /// <param name="resource">The dataset resource representation to use for the patch. Only fields present in the resource will be updated.</param>
+        /// <param name="options">The options for the operation. May be null, in which case defaults will be supplied.</param>
+        /// <returns></returns>
+        public virtual BigQueryDataset PatchDataset(string projectId, string datasetId, Dataset resource, PatchDatasetOptions options = null) =>
+            PatchDataset(GetDatasetReference(projectId, datasetId), resource, options);
+
+        /// <summary>
+        /// Patches the specified dataset with fields in the given resource.
```

Comment:
Should this use "...within this client's project.." like the associated update overload?